crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -217,6 +218,10 @@ where
                         debug!("truncated response received, retrying over TCP");
                         Ok(response)
                     }
+                    Err(e) if matches!(e.kind(), ProtoErrorKind::NoRecordsFound{ response_code, .. } if *response_code == ResponseCode::NXDomain ) => {
+                        debug!("NoRecordsFound with NXDomain, not retrying over TCP");
+                        return Err(e).map_err(ProtoError::from)
+                    }
```

Comment:
I think this case can be rewritten as:

```rust
Err(e) if matches!(e.kind(), ProtoErrorKind::NoRecordsFound{ response_code: ResponseCode::NXDomain, .. } ) => {
    debug!("NoRecordsFound with NXDomain, not retrying over TCP");
    return Err(e);
}
```

it matches the `ProtoErrorKind::NoRecordsFound` variant more directly.

Can the statement `return Err(e).map_err(ProtoError::from)` be shortened to `Err(e);`, because `e` is already of type `ProtoError`.